lib/pi_piper.rb

```diff
@@ -1,8 +1,39 @@
 require 'eventmachine'
-Dir[File.dirname(__FILE__) + '/pi_piper/*.rb'].each {|file| require file unless file.end_with?('bcm2835.rb') }
+
+require 'pi_piper/version'
+require 'pi_piper/frequency'
+require 'pi_piper/driver'
+require 'pi_piper/i2c'
+require 'pi_piper/pin'
+require 'pi_piper/spi'
+require 'pi_piper/pwm'
 
 module PiPiper
+  class PinError < StandardError; end
```

Comment:
In fact we do not need this error class.
- Errors are now raised in drivers not in Pin
- Errors are only common ArgumentError, NoMethodError, NotImplementedError, and should stay so for better debugging IMHO.